src/Operators/finitedifference.jl

```diff
@@ -1588,6 +1588,209 @@ end
     return Geometry.Contravariant3Vector(vᶠ)
 end
 
+
+
+#########################
+"""
+    U = FCTZalesak(;boundaries)
+    U.(A, Φ, Φᵗᵈ)
+
+Correct the flux using the flux-corrected transport formulation by Zalesak
+Input arguments:
+- a face-valued vector field `A`
+- a center-valued field `Φ`
+- a center-valued field `Φᵗᵈ`
+```math
+Φ_j^{n+1} = Φ_j^{td} - \frac{dt}{dx}(C_{j+\frac{1}{2}}A_{j+\frac{1}{2}} - C_{j-\frac{1}{2}}A_{j-\frac{1}{2}})
+```
+This stencil is based on [zalesak1979fully](@cite).
+
+Supported boundary conditions are:
+- [`ThirdOrderOneSided(x₀)`](@ref): uses the third-order downwind reconstruction to compute `x` on the left boundary,
+and the third-order upwind reconstruction to compute `x` on the right boundary.
+
+!!! note
+    Similar to the [`Upwind3rdOrderBiasedProductC2F`](@ref) operator, these boundary conditions do not define 
+    the value at the actual boundary faces, and so this operator cannot be materialized directly: it needs to 
+    be composed with another operator that does not make use of this value, e.g. a [`DivergenceF2C`](@ref) operator, 
+    with a [`SetValue`](@ref) boundary.
+"""
+struct FCTZalesak{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+FCTZalesak(; kwargs...) = FCTZalesak(NamedTuple(kwargs))
+
+return_eltype(::FCTZalesak, A, Φ, Φᵗᵈ) =
+    Geometry.Contravariant3Vector{eltype(eltype(A))}
+
+return_space(
+    ::FCTZalesak,
+    A_space::Spaces.FaceFiniteDifferenceSpace,
+    Φ_space::Spaces.CenterFiniteDifferenceSpace,
+    Φᵗᵈ_space::Spaces.CenterFiniteDifferenceSpace,
+) = A_space
+return_space(
+    ::FCTZalesak,
+    A_space::Spaces.FaceExtrudedFiniteDifferenceSpace,
+    Φ_space::Spaces.CenterExtrudedFiniteDifferenceSpace,
+    Φᵗᵈ_space::Spaces.CenterExtrudedFiniteDifferenceSpace,
+) = A_space
+
+@inline function fct_zalesak(
+    Aⱼ₋₁₂,
+    Aⱼ₊₁₂,
+    Aⱼ₊₃₂,
+    ϕⱼ₋₁,
+    ϕⱼ,
+    ϕⱼ₊₁,
+    ϕⱼ₊₂,
+    ϕⱼ₋₁ᵗᵈ,
+    ϕⱼᵗᵈ,
+    ϕⱼ₊₁ᵗᵈ,
+    ϕⱼ₊₂ᵗᵈ,
+    dxⱼ,
+    dxⱼ₊₁,
+)
+    # 1/dt is in ϕⱼ₋₁, ϕⱼ, ϕⱼ₊₁, ϕⱼ₊₂, ϕⱼ₋₁ᵗᵈ, ϕⱼᵗᵈ, ϕⱼ₊₁ᵗᵈ, ϕⱼ₊₂ᵗᵈ
+
+    stable_zero = zero(eltype(Aⱼ₊₁₂))
+
+    if (
+        Aⱼ₊₁₂ * (ϕⱼ₊₁ᵗᵈ - ϕⱼᵗᵈ) < stable_zero ||
+        Aⱼ₊₁₂ * (ϕⱼ₊₂ᵗᵈ - ϕⱼ₊₁ᵗᵈ) < stable_zero ||
+        Aⱼ₊₁₂ * (ϕⱼᵗᵈ - ϕⱼ₋₁ᵗᵈ) < stable_zero
+    )
+        Aⱼ₊₁₂ = stable_zero
+    end
+    ϕⱼᵐᵃˣ = max(ϕⱼ₋₁, ϕⱼ, ϕⱼ₊₁, ϕⱼ₋₁ᵗᵈ, ϕⱼᵗᵈ, ϕⱼ₊₁ᵗᵈ)
+    ϕⱼᵐⁱⁿ = min(ϕⱼ₋₁, ϕⱼ, ϕⱼ₊₁, ϕⱼ₋₁ᵗᵈ, ϕⱼᵗᵈ, ϕⱼ₊₁ᵗᵈ)
+    Pⱼ⁺ = max(stable_zero, Aⱼ₋₁₂) - min(stable_zero, Aⱼ₊₁₂)
+    Qⱼ⁺ = (ϕⱼᵐᵃˣ - ϕⱼᵗᵈ) * dxⱼ
+    Rⱼ⁺ = (Pⱼ⁺ > stable_zero ? min(1, Qⱼ⁺ / Pⱼ⁺) : stable_zero)
```

Comment:
Can we also use `one(::Type)` instead of `1` here?

Reply:
just updated